frontend/models/product.ts

```diff
@@ -90,10 +100,46 @@ function getVariants(shopifyProduct: NonNullable<FindProductQuery['product']>) {
          specifications: variant.specifications?.value ?? null,
          warranty: variant.warranty?.value ?? null,
          crossSellVariants: getCrossSellVariants(variant),
+         disableWhenOOS: variant.disableWhenOOS?.value === 'true',
       };
    });
 }
 
+function getImages(
+   shopifyImages: NonNullable<FindProductQuery['product']>['images'],
+   variants: ReturnType<typeof getVariants>
+) {
+   return shopifyImages.nodes.filter(
+      (image) =>
+         !variants.find(
+            (variant) =>
+               image.altText &&
+               variant.sku === image.altText &&
+               variant.disableWhenOOS &&
+               variant.quantityAvailable != null &&
+               variant.quantityAvailable <= 0
```

Comment:
If we pass in only active variants, then we probably don't need these comparisons I think.

Reply:
See [answer above](https://github.com/iFixit/react-commerce/pull/774#discussion_r988575778)